2011-02-16  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* stdio-common/psiginfo.c (psiginfo): Check pinfo->si_signo against
	SIGRTMIN and SIGRTMAX and print information in that case only when
	SIGRTMIN is defined.

---
 stdio-common/psiginfo.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)
--- a/stdio-common/psiginfo.c
+++ b/stdio-common/psiginfo.c
@@ -86,8 +86,12 @@
   const char *desc;
   if (pinfo->si_signo >= 0 && pinfo->si_signo < NSIG
       && ((desc = INTUSE(_sys_siglist)[pinfo->si_signo]) != NULL
-	  || (pinfo->si_signo >= SIGRTMIN && pinfo->si_signo < SIGRTMAX)))
+#ifdef SIGRTMIN
+	  || (pinfo->si_signo >= SIGRTMIN && pinfo->si_signo < SIGRTMAX)
+#endif
+	 ))
     {
+#ifdef SIGRTMIN
       if (desc == NULL)
 	{
 	  if (pinfo->si_signo - SIGRTMIN < SIGRTMAX - pinfo->si_signo)
@@ -106,6 +110,7 @@
 	    }
 	}
       else
+#endif
 	fprintf (fp, "%s (", _(desc));
 
       const char *base = NULL;
